jwk.go

```diff
@@ -336,6 +350,34 @@ func (s *JSONWebKeySet) Key(kid string) []JSONWebKey {
 	return keys
 }
 
+func (s *JSONWebKeySet) UnmarshalJSON(data []byte) (err error) {
+	type rawJSONWebKeySet struct {
+		Keys []json.RawMessage `json:"keys"`
+	}
+
+	var rs rawJSONWebKeySet
+	err = json.Unmarshal(data, &rs)
+	if err != nil {
+		return err
+	}
+
+	for _, rk := range rs.Keys {
+		var k JSONWebKey
+		err = json.Unmarshal(rk, &k)
+		if err != nil {
+			// Skip key and continue unmarshalling the key set if key unmarshal
+			// failed because of unsupported key type or parameters.
+			if !errors.Is(err, errUnsupportedJWK) {
+				return err
+			}
+		} else {
+			s.Keys = append(s.Keys, k)
```

Comment:
[The docs for Unmarshal](https://pkg.go.dev/encoding/json#Unmarshal) say:

> To unmarshal a JSON array into a slice, Unmarshal resets the slice length to zero and then appends each element to the slice.

We should do the same here, resetting `s.Keys` to zero length just in case we wind up unmarshaling into a pre-existing JSONWebKeySet.